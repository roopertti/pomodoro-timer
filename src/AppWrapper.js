import React from 'react'
import PropTypes from 'prop-types'

import { StoreProvider } from 'Contexts/StoreContext'

function AppWrapper ({ children }) {
  return (
    <StoreProvider>
      {children}
    </StoreProvider>
  )
}

AppWrapper.propTypes = {
  children: PropTypes.node
}

export default AppWrapper
