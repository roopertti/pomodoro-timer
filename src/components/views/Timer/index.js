import React from 'react'

function Timer () {
  return (
    <div>Timer component placeholder</div>
  )
}

export default Timer
