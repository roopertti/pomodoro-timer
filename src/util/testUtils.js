import { render } from '@testing-library/react'
import AppWrapper from '../AppWrapper'

function renderWithWrappers (ui, options) {
  return render(
    ui,
    {
      wrapper: AppWrapper,
      ...options
    }
  )
}

export * from '@testing-library/react'

export { renderWithWrappers as render }
