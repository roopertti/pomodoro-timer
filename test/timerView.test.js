import React from 'react'
import { act } from 'react-dom/test-utils'
import renderer from 'react-test-renderer'

import AppWrapper from '../src/AppWrapper'
import Timer from 'Components/views/Timer'
import { render, fireEvent, waitFor, cleanup } from 'Util/testUtils'

describe('Timer view integration tests', () => {
  it('renders Timer view and compares snapshot', () => {
    const tree = renderer.create(
      (
        <AppWrapper>
          <Timer />
        </AppWrapper>
      )
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders Timer, start work timer and pause timer', async () => {
    const { getByTestId } = render(<Timer />)
    expect(getByTestId('timer-view')).toBeInTheDocument()
    fireEvent.click(getByTestId('start-work'))
    await waitFor(() => {
      expect(getByTestId('timer-mmss')).toHaveTextContent('25:00')
    })
    expect(getByTestId('timer-btn')).toHaveTextContent('Stop')
    fireEvent.click(getByTestId('timer-btn'))
    expect(getByTestId('timer-btn')).toHaveTextContent('Resume')
  })
})
