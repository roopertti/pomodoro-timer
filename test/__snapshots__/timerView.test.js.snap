// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer view integration tests renders Timer view and compares snapshot 1`] = `
<section
  data-testid="timer-view"
  style={
    Object {
      "opacity": 0,
      "transform": "translateY(-30px)",
    }
  }
>
  <div
    style={
      Object {
        "height": "100%",
        "opacity": 1,
      }
    }
  >
    <header
      className="py-4 mb-4"
    >
      <h2>
        Select your next activity
      </h2>
      <p
        className="text-center"
      >
        Either start working or have a short break.
      </p>
    </header>
    <div
      className="block sm:flex"
    >
      <div
        className="mb-4 sm:mr-2"
      >
        <button
          className="btn btn-success"
          data-testid="start-work"
          onClick={[Function]}
          style={
            Object {
              "transform": "translateY(0px)",
            }
          }
        >
          Start working (25min)
        </button>
      </div>
      <div
        className="mb-4"
      >
        <button
          className="btn btn-secondary"
          data-testid="start-break"
          onClick={[Function]}
          style={
            Object {
              "transform": "translateY(0px)",
            }
          }
        >
          Start break (5min)
        </button>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "height": "0",
        "opacity": 0,
      }
    }
  >
    <p
      className="flex justify-center"
    >
      Work in progress...
    </p>
    <div
      className="flex justify-center mb-4"
      data-testid="timer-component"
    >
      <span
        className="timer"
        data-testid="timer-mmss"
      >
        00:00
      </span>
    </div>
    <div
      className="flex justify-center mb-4"
    >
      <button
        className="btn btn-success"
        data-testid="timer-btn"
        onClick={[Function]}
        style={
          Object {
            "transform": "translateY(0px)",
          }
        }
      >
        Resume
      </button>
    </div>
  </div>
</section>
`;
